### PR TITLE
luks2: add DetectCryptsetupFeatures

### PR DESCRIPTION
**Pull request by chrisccoulson**
We depend on some cryptsetup features that aren't in all of the
cryptsetup versions we have in Ubuntu, so add a simple internal
feature detection API for those. This is motivated by the fact
that we're going to depend on --token-replace, which has just been
approved to land upstream and which we'll need to backport.

### CODE REVIEW

File: internal/luks2/cryptsetup.go [hunk 1]
```diff
@@ -76,6 +107,34 @@ func cryptsetupCmd(stdin io.Reader, callback func(cmd *exec.Cmd) error, args ...
...
+			if n == 3 {
+				if major >= 3 || (major == 2 && minor >= 1) {
+					features |= FeatureHeaderSizeSetting
+				}
+				if major >= 3 || (major == 2 && minor >= 1) || (major == 2 && minor == 0 && patch >= 3) {
+					features |= FeatureTokenImport
+				}
+			}
+		}
+		if err := cryptsetupCmd(nil, nil, "--test-args", "token", "import", "--token-id", "0",
```
Comment (pedronis) [inline internal/luks2/cryptsetup.go:130]: is --test-args an undocumented (I don't see it in the man page) but supported option?

**Review by pedronis — COMMENTED** [review]
a question

File: internal/luks2/cryptsetup.go (hunk 1 above)
Comment (chrisccoulson) [reply internal/luks2/cryptsetup.go:130]: It is undocumented. It's currently used by cryptsetup's unit tests for validating combinations of commandline arguments.

Comment (chrisccoulson) [reply internal/luks2/cryptsetup.go:130]: It is documented in the `--help` output though:
```
      --test-args                       Do not run action, just validate all
                                        command line parameters
```

**Review by pedronis — APPROVED** [review]
+1

**Pull request merged**